src/dex/swaap-v1/swaap-v1.ts

```diff
@@ -0,0 +1,621 @@
+import {
+  SwaapV1Config,
+  Adapters,
+  SUBGRAPH_FETCH_ALL_POOOLS_RQ,
+  SUBGRAPH_TIMEOUT,
+  MAX_POOL_CNT,
+  POOL_CACHE_TTL,
+  MAX_GAS_COST_ESTIMATION,
+  RECCURING_POOL_FETCH_INTERVAL_MS,
+} from './config';
+import { SwaapV1Pool } from './swaap-v1-pool';
+import PoolABI from '../../abi/swaap-v1/pool.json';
+import { PoolQuotations } from './libraries/PoolQuotations';
+
+export class SwaapV1 extends SimpleExchange implements IDex<SwaapV1Data> {
+  static readonly poolInterface = new Interface(PoolABI);
+
+  protected config: DexParams;
+  protected subgraphURL: string;
+  protected exchangeProxy: string;
+
+  protected eventPools: { [poolAddress: string]: SwaapV1Pool } = {};
+
+  readonly hasConstantPriceLargeAmounts = false;
+  readonly isFeeOnTransferSupported: boolean = false;
+
+  public static dexKeysWithNetwork: { key: string; networks: Network[] }[] =
+    getDexKeysWithNetwork(SwaapV1Config);
+
+  logger: Logger;
+
+  constructor(
+    readonly network: Network,
+    readonly dexKey: string,
+    readonly dexHelper: IDexHelper,
+    protected adapters = Adapters[network], // TODO: add any additional optional params to support other fork DEXes
+  ) {
+    super(dexHelper, dexKey);
+    this.config = SwaapV1Config[dexKey][network];
+    this.logger = dexHelper.getLogger(`${dexKey}-${network}`);
+    this.subgraphURL = this.config.subgraphURL;
+    this.exchangeProxy = this.config.exchangeProxy;
+  }
+
+  // Initialize pricing is called once in the start of
+  // pricing service. It is intended to setup the integration
+  // for pricing requests. It is optional for a DEX to
+  // implement this function
+  async initializePricing(blockNumber: number) {
+    await this.getPools(blockNumber, true);
+    setInterval(
+      () => this.getPools(null, false),
+      RECCURING_POOL_FETCH_INTERVAL_MS,
+    );
+  }
+
+  async getPools(_blockNumber: number | null, init: boolean) {
+    const blockNumber = _blockNumber
+      ? _blockNumber
+      : await this.dexHelper.provider.getBlockNumber();
+    const allPools = await this.fetchAllSubgraphPools(blockNumber);
+    if (!allPools)
+      if (init) {
+        throw new Error(
+          'initializePricing: SwaapV1 cfgInfo still null after init',
+        );
+      }
+    for (const poolConfig of allPools.values()) {
+      if (
+        init ||
+        !Object.values(this.eventPools)
+          .map(e => e.id)
+          .includes(poolConfig.id)
+      ) {
+        await this.addPool(poolConfig, blockNumber);
+      }
+    }
+  }
+
+  async addPool(poolConfig: SubgraphPoolBase, blockNumber: number) {
+    const pool = new SwaapV1Pool(
+      this.dexKey,
+      this.network,
+      poolConfig,
+      this.dexHelper,
+    );
+    const state = await pool.generateState(blockNumber);
+    pool.setState(state, blockNumber);
+
+    this.dexHelper.blockManager.subscribeToLogs(
+      pool,
+      pool.addressSubscribers,
+      blockNumber,
+    );
```

Comment:
Can migrate generate/set state and subscribe to logs to using new `initialize` function instead.  It looks like `addressSubscribers` needs renamed everywhere for that to work.